app/views/users/_matches_list.html.haml

```diff
@@ -18,3 +18,5 @@
         .match-right
           .match-topic
             = image_tag "#{target.topic.title}x2.png", class: 'match-topic-img'
+          .unread-msgs{ class: ('hidden' if match.chat.unread_messages_count(user.id) == 0) }
```

Comment:
In order to check whether a associations collection is empty or not, it's better to use the `exists?` method in this case, due to the fact that it won't trigger a `count` on the DB. Instead, the `exists?` method will use `LIMIT 1`, which is faster. So, make `unread_messages` public, and use it here: 
`.unread-msgs{ class: ('hidden' unless match.chat.unread_messages(user.id).exists?) }`

You should read the following article, it explains how `empty?`, `any?`, `blank?` and `exists?` work.
http://andreigridnev.com/empty-blank-any-exists-methods-of-ruby-on-rails-activerecord/